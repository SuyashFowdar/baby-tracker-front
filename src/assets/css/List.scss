.list {
  .items {
    margin: 10px;
    border-radius: 5px;
    background-color: rgb(255, 255, 255);
    border: 1px solid rgb(180, 180, 180);

    & > .link:nth-child(even) {
      background-color: rgba(0, 150, 255, 0.1);

      &:hover {
        background-color: rgba(0, 0, 0, 0.2);
      }
    }
  }

  .item {
    font-size: 20px;
    padding: 10px;
  }

  input {
    width: calc(50% - 20px);
  }
}
