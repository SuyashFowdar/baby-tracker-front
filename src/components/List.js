import React from 'react';
import PropTypes from 'prop-types';
import '../assets/css/List.scss';

const List = ({
  list,
  itemKey,
  displayAttr,
  toAddAttr,
  addItem,
  unit,
}) => (
  <div className="list">
    <div className="items">
      {list.map((item) => (
        <div key={item[itemKey]} className="row link">
          {displayAttr.map((attr) => (
            <div key={attr} className="flex item">
              <div>{`${item[attr]} ${attr === 'amount' ? unit : ''}`}</div>
            </div>
          ))}
        </div>
      ))}
    </div>
    <form onSubmit={addItem} className="row main-center wrap">
      {toAddAttr.map((attr) => (
        <input key={attr} type="text" name={attr} placeholder={attr} />
      ))}
      <input type="submit" value="Add New" className="link" />
    </form>
  </div>
);

List.defaultProps = {
  list: [],
  unit: '',
};

List.propTypes = {
  list: PropTypes.instanceOf(Array),
  unit: PropTypes.string,
  itemKey: PropTypes.string.isRequired,
  addItem: PropTypes.func.isRequired,
  displayAttr: PropTypes.instanceOf(Array).isRequired,
  toAddAttr: PropTypes.instanceOf(Array).isRequired,
};

export default List;
