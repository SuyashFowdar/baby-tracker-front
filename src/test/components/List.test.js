import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { BrowserRouter as Router } from 'react-router-dom';
import List from '../../components/List';
import reducer from '../../reducers/index';

it('renders correctly', () => {
  const store = createStore(reducer);
  const tree = renderer
    .create((
      <Provider store={store}>
        <Router><List itemKey="" addItem={() => {}} displayAttr={[]} toAddAttr={[]} /></Router>
      </Provider>
    ))
    .toJSON();
  expect(tree).toMatchSnapshot();
});
