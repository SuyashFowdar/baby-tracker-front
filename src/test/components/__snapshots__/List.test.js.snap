// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="list"
>
  <div
    className="items"
  />
  <form
    className="row main-center wrap"
    onSubmit={[Function]}
  >
    <input
      className="link"
      type="submit"
      value="Add New"
    />
  </form>
</div>
`;
